mono/metadata/mono-perfcounters.c

```diff
@@ -1483,276 +1567,438 @@ typedef struct {
 	int type;
 } CounterCreationData;
 
+TYPED_HANDLE_DECL (CounterCreationData);
+
+static char*
+append (char* cursor, const char* str)
+{
+	int const length = strlen (str) + 1;
+	return (char*)memcpy (cursor, str, length) + length;
+}
+
+static void
+ves_icall_System_Diagnostics_PerformanceCounterCategory_Create_create_temp (
+	MonoArrayHandle items, int i, char **counter_info, MonoError *error)
+{ // function to avoid creating handles in loop (of user-controlled count)
+
+	HANDLE_FUNCTION_ENTER()
+
+	MONO_HANDLE_LOCAL_VARIABLE_INITIALIZED_NULL (CounterCreationData, data);
+	MONO_HANDLE_ARRAY_GETREF (data, items, i);
+	// FIXME All that should really happen here is compute the size of the UTF8 conversion,
+	// not actually allocate and copy the data.
+	counter_info [i * 2] = mono_string_handle_to_utf8 (MONO_HANDLE_NEW_GET (MonoString, data, name), error);
+	goto_if_nok (error, failure);
+	counter_info [i * 2 + 1] = mono_string_handle_to_utf8 (MONO_HANDLE_NEW_GET (MonoString, data, help), error);
+	goto_if_nok (error, failure);
+failure:
+	HANDLE_FUNCTION_RETURN();
+}
+
+static char*
+ves_icall_System_Diagnostics_PerformanceCounterCategory_Create_copy_temp_to_shared (
```

Comment:
Same as above: ok to use a short name, best not to use a `ves_icall_`-style prefix.